src/index.ts

```diff
@@ -257,8 +285,8 @@ export default class SwupScrollPlugin extends Plugin {
 		}
 
 		// Try scrolling to a given anchor
-		const scrollTarget = visit.scroll.target || visit.to.hash;
-		if (this.maybeScrollToAnchor(scrollTarget, this.shouldAnimate('betweenPages'))) {
+		const scrollTarget = visit.scroll.target ?? visit.to.hash;
```

Comment:
Fix same [issue](https://github.com/swup/swup/pull/768) fixed in swup core: allow setting `scroll.target` to `false` to avoid anchor scrolling on history visits.